app/frontend/src/components/UploadFile/UploadFile.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useState, ChangeEvent, FormEvent } from "react";
+import styles from "./UploadFile.module.css";
+import { IIconProps, Callout, ActionButton, PrimaryButton, Label, IconButton, Text } from "@fluentui/react";
+import { SimpleAPIResponse, uploadFileApi, deleteUploadedFileApi, listUploadedFilesApi } from "../../api";
+import { useMsal } from "@azure/msal-react";
+import { useLogin, getToken } from "../../authConfig";
+
+interface Props {
+    className?: string;
+}
+
+export const UploadFile: React.FC<Props> = ({ className }: Props) => {
+    // State variables to manage the component behavior
+    const [isCalloutVisible, setIsCalloutVisible] = useState<boolean>(false);
+    const [isUploading, setIsUploading] = useState<boolean>(false);
+    const [isLoading, setIsLoading] = useState<boolean>(true);
+    const [uploadedFile, setUploadedFile] = useState<SimpleAPIResponse>();
+    const [uploadedFiles, setUploadedFiles] = useState<string[]>([]);
+
+    if (!useLogin) {
+        console.error("This component requires useLogin to be true");
+        return null;
+    }
+
+    const client = useMsal().instance;
+
+    // Handler for the "Manage file uploads" button
+    const handleButtonClick = async () => {
+        setIsCalloutVisible(!isCalloutVisible); // Toggle the Callout visibility
+
+        // Update uploaded files by calling the API
+        try {
+            const idToken = await getToken(client);
+            if (!idToken) {
+                throw new Error("No authentication token available");
+            }
+            listUploadedFilesApi(idToken).then(files => setUploadedFiles(files));
+            setIsLoading(false);
+        } catch (error) {
+            console.error(error);
+            setIsLoading(false);
+        }
+    };
+
+    const handleRemoveFile = async (filename: string) => {
+        try {
+            const idToken = await getToken(client);
+            if (!idToken) {
+                throw new Error("No authentication token available");
+            }
+            await deleteUploadedFileApi(filename, idToken);
+            const newFiles = uploadedFiles.filter(file => file !== filename);
+            // To discuss: We could also poll the list endpoint here
```

Comment:
IMO it's OK to immediately show the file deleted. We can file a follow up issue if we don't like this behavior later on?

Reply:
I've decided it's easier to do the Right Thing than deal with possible confusion from an over-optimistic UI. Now I've got it so that it shows a status for deletion next to each file like "Deleting..." "File deleted!" and then finally the file goes away when the new list comes back. Also shows an error status if there's an issue deleting the file.